tests/base.go

```diff
@@ -73,78 +64,80 @@ func RemoteRun(user string, addr string, cmd string) (string, error) {
 	// Connect
 	client, err := ssh.Dial("tcp", net.JoinHostPort(addr, "22"), config)
 	if err != nil {
-		return "", err
+		return "", errors.Wrapf(err, "error starting ssh connection ")
 	}
 	// Create a session. It is one session per command.
 	session, err := client.NewSession()
 	if err != nil {
-		return "", err
+		return "", errors.Wrapf(err, "error creating new session")
 	}
 	defer session.Close()
 	var b bytes.Buffer  // import "bytes"
 	session.Stdout = &b // get output
 	err = session.Run(cmd)
-	return b.String(), err
+	if err != nil {
+		return "", errors.Wrapf(err, "error excuting command on remote")
+	}
+	return b.String(), nil
 }
 
-func VerifyIPs(wgConfig string, verifyIPs []string) bool {
-	UpWg(wgConfig)
+// TODO: investigate if this is needed
+func VerifyIPs(wgConfig string, verifyIPs []string, sk string) error {
```

Comment:
the name is confusing, what does verifyIPs mean in this context?
1- it should be `VerifyIP` singular that does whatever the check is, and you can have another function executing the same thing using goroutines on the list
2- the function seems to be about the reachability, so should be isIPReachable instead
3- the call invokes ping from local machine against the IP set, that completely ignores the err variable?

4- we do the same check again but invoking ssh, on every IP.

Checking with ICMP packets isn't always the right thing to do just for connectivity, while SSH should be guaranteed to work if you have access on the machines, so it's redundant in my opinion, if anything you can stick to the SSH checks only, and if you can start an SSH session, that implies you can reach the machine